scipy/stats/_binomtest.py

```diff
@@ -0,0 +1,314 @@
+        return s
+
+    def proportion_ci(self, confidence_level=0.95, method='exact'):
+        """
+        Compute the confidence interval for the estimated proportion.
+
+        Parameters
+        ----------
+        confidence_level : float, optional
+            Confidence level for the computed confidence interval
+            of the estimated proportion. Default is 0.95.
+        method : {'exact', 'wilson', 'wilsoncc'}, optional
+            Selects the method used to compute the confidence interval
+            for the estimate of the proportion:
+
+            'exact' :
+                Use the Clopper-Pearson exact method [1]_.
+            'wilson' :
+                Wilson's method, without continuity correction ([2]_, [3]_).
+            'wilsoncc' :
+                Wilson's method, with continuity correction ([2]_, [3]_).
+
+            Default is ``'exact'``.
+
+        Returns
+        -------
+        ci : namedtuple with fields ``low`` and ``high``
+            Contains the lower and upper bounds of the confidence interval.
+
+        References
+        ----------
+        .. [1] C. J. Clopper and E. S. Pearson, The use of confidence or
+               fiducial limits illustrated in the case of the binomial,
+               Biometrika, Vol. 26, No. 4, pp 404-413 (Dec. 1934).
+        .. [2] E. B. Wilson, Probable inference, the law of succession, and
+               statistical inference, J. Amer. Stat. Assoc., 22, pp 209-212
+               (1927).
+        .. [3] Robert G. Newcombe, Two-sided confidence intervals for the
+               single proportion: comparison of seven methods, Statistics
+               in Medicine, 17, pp 857-872 (1998).
+
+        Examples
+        --------
+        >>> from scipy.stats import binomtest
+        >>> result = binomtest(k=7, n=50, p=0.1)
+        >>> result.proportion_estimate
+        0.14
+        >>> result.proportion_ci
+        ConfidenceInterval(low=0.04246878737883377, high=0.30910696596481874)
+        """
+        if method not in ('exact', 'wilson', 'wilsoncc'):
+            raise ValueError("method must be one of 'exact', 'wilson' or "
+                             "'wilsoncc'.")
+        if not (0 <= confidence_level <= 1):
+            raise ValueError('confidence_level must be in the interval '
+                             '[0, 1].')
+        if method == 'exact':
+            low, high = _binom_exact_conf_int(self.k, self.n,
+                                              confidence_level,
+                                              self.alternative)
+        else:
+            # method is 'wilson' or 'wilsoncc'
+            low, high = _binom_wilson_conf_int(self.k, self.n,
+                                               confidence_level,
+                                               self.alternative,
+                                               correction=method == 'wilsoncc')
+        return ConfidenceInterval(low=low, high=high)
```

Comment:
That scenario is possible, but the same argument could be applied to the return value of *any* function in SciPy, and we generally don't include input parameters in the output.  E.g. we don't include `ddof` in the output of `chisquare` or `power_divergence`, and we don't include `equal_var` or `alternative` in the output of `ttest_ind`.

The only reason `BinomTestResult` saves some of the input parameters is because they are needed for the `proportion_ci()` method.  There are no methods on the `ConfidenceInterval` object that would require saving copies of the input parameters.

Having said that, in my most recent updates, I changed `ConfidenceInterval` from a `namedtuple` to a class (decorated as a [`dataclass`](https://docs.python.org/3/library/dataclasses.html) for convenience), which will make it easier to add attributes to the object if we decide to in the future.

Reply:
>That scenario is possible, but the same argument could be applied to the return value of any function in SciPy, and we generally don't include input parameters in the output...
The only reason BinomTestResult saves some of the input parameters is because they are needed for the proportion_ci() method. There are no methods on the ConfidenceInterval object that would require saving copies of the input parameters.

Ok, consistency is a reasonable argument here, that might be more of an issue in a cluster where you'd want to pass the state around some. In that case the scipy-user will need to code out their stuff custom. 

>Having said that, in my most recent updates, I changed ConfidenceInterval from a namedtuple to a class (decorated as a dataclass for convenience), which will make it easier to add attributes to the object if we decide to in the future.

+1 for `dataclass`es, I've used them in some personal web projects, they're pretty handy.